### PR TITLE
Improve the handling of targets in the Swift xtask.

### PR DESCRIPTION
**Pull request by pixlwave**
Currently there are 3 separate code paths to build an xcframework depending on if you specify `--only-target`, nothing (all the targets) or `--sequentially` (all the targets one-by-one aka Stefan mode). This PR fixes that by defining all of the known targets and using the input to pick which ones should be included. This paves the way forward for us to include more supported platforms in the task (e.g. Mac Catalyst) whilst also being able to build for a certain subset of the targets.

### CODE REVIEW

**Review by andybalaam — APPROVED** [review]
Looks good, thanks!

File: xtask/src/swift.rs [hunk 1]
```diff
@@ -66,7 +69,68 @@ impl SwiftArgs {
...
+            Platform::Ios => "libmatrix_sdk_ffi_ios.a",
+            Platform::IosSimulator => "libmatrix_sdk_ffi_iossimulator.a",
+        }
+    }
+}
+
+/// The base name of the FFI library.
 const FFI_LIBRARY_NAME: &str = "libmatrix_sdk_ffi.a";
+/// The list of targets supported by the SDK.
+const TARGETS: &[Target] = &[
```
Comment (stefanceriu) [inline xtask/src/swift.rs:111]: Can we please define Catalyst as well while we're at it?

File: xtask/src/swift.rs [hunk 2]
```diff
@@ -66,7 +69,68 @@ impl SwiftArgs {
...
+    Target { triple: "aarch64-apple-ios", platform: Platform::Ios, description: "iOS" },
```
Comment (stefanceriu) [inline xtask/src/swift.rs:112]: Don't forget to reindent this one

File: xtask/src/swift.rs [hunk 3]
```diff
@@ -66,7 +69,68 @@ impl SwiftArgs {
...
+    Target {
+        triple: "aarch64-apple-darwin",
```
Comment (stefanceriu) [inline xtask/src/swift.rs:114]: I wonder if we should define the triples as an enum as well

File: xtask/src/swift.rs [hunk 4]
```diff
@@ -276,6 +253,81 @@ fn build_xcframework(
     Ok(())
 }
 
+/// Builds the SDK for the specified targets and profile.
+fn build_targets(
+    targets: Vec<&Target>,
+    profile: &str,
+    sequentially: bool,
+) -> Result<HashMap<Platform, Vec<Utf8PathBuf>>> {
+    if sequentially {
```
Comment (stefanceriu) [inline xtask/src/swift.rs:262]: We can get rid of this if we go with only building for iOS and simulators.

**Review by stefanceriu — APPROVED** [review]
Very very nice, thank you!

File: xtask/src/swift.rs (hunk 2 above)
Comment (pixlwave) [reply xtask/src/swift.rs:112]: I don't think I can. Won't fmt put it back again?

File: xtask/src/swift.rs (hunk 1 above)
Comment (pixlwave) [reply xtask/src/swift.rs:111]: Will do in a follow-up PR

File: xtask/src/swift.rs (hunk 4 above)
Comment (pixlwave) [reply xtask/src/swift.rs:262]: Along with this.

File: xtask/src/swift.rs (hunk 3 above)
Comment (pixlwave) [reply xtask/src/swift.rs:114]: I would but I'm not sure Rust has the same concept as Swift where you can choose the raw value type of an Enum. Looks like it's always an `isize`. 🤷‍♂️

File: xtask/src/swift.rs (hunk 2 above)
Comment (stefanceriu) [reply xtask/src/swift.rs:112]: OMG that's horrible 🤦‍♂️

File: xtask/src/swift.rs (hunk 3 above)
Comment (stefanceriu) [reply xtask/src/swift.rs:114]: Alrighty, leave it alone then ¯\_(ツ)_/¯

**Pull request merged**